spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -501,24 +502,24 @@ else if (result instanceof Message) {
 		}
 	}
 
-	private Message<?> checkHeaders(Object result, String topic, @Nullable Object source) { // NOSONAR (complexity)
-		Message<?> reply = (Message<?>) result;
+	private Message<?> checkHeaders(Message<?> reply, @Nullable String topic, @Nullable Object source) { // NOSONAR (complexity)
```

Comment:
I think choosing `Message<?>` as the first argument type makes more sense, for this method is only meant to be used with that precondition.

Reply:
not sure whether we can use `Nullable` to `topic`, but from its invocation context, seems we could never rule out the nullness possibility?